src/PlatformCompatibilityTests/ServiceControlCompatibilityTests/SuccessfulRetryTests.cs

```diff
@@ -0,0 +1,74 @@
+﻿namespace ServiceControlCompatibilityTests
+{
+    using Autofac;
+    using NServiceBus;
+    using NUnit.Framework;
+    using System;
+    using System.Threading.Tasks;
+
+    [TestFixture]
+    class SuccessfulRetryTests : SqlScTest
+    {
+        [TestCaseSource(nameof(AllTransports))]
+        public async Task Can_successfully_retry_a_failed_message(Type transportDetailType)
+        {
+            var transportDetails = StartUp(transportDetailType);
+            await RunTest(transportDetails);
+        }
+
+        async Task RunTest(ITransportDetails transportDetails)
+        {
+            var handlerInstance = new TestMessageHandler();
+            var builder = new ContainerBuilder();
+            builder.RegisterInstance(handlerInstance);
+            var container = builder.Build();
+
+            var endpointA = await CreateSqlEndpoint("EndpointA", transportDetails, container);
+            await CreateSqlEndpoint("EndpointB", transportDetails, container);
+
+            handlerInstance.ShouldFail = true;
+            endpointA.Send("EndpointB", new TestMessage());
+
+            var failedMessage = await ServiceControl.WaitForNewFailingMessages("EndpointA");
+
+            handlerInstance.ShouldFail = false;
+            await ServiceControl.RetryMessageId(failedMessage.Id);
+
+            var handled = handlerInstance.WaitForSuccessfulMessage().Wait(60000); // Is this a reasonable time to wait?
+
+            Assert.IsTrue(handled, "Did not process the retry successfully within the time limit");
```

Comment:
Maybe also log the time we waited (60 secs) that might come in handy when a test fails and we trace the build/test run logs.